code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -98,54 +98,27 @@
 	body_parts_covered = FACE|EYES
 	action_button_name = "Toggle MUI"
 	origin_tech = "{'programming':5,'engineering':5}"
-	var/active = FALSE
-	var/mob/observer/eye/freelook/cameranet/eye
 
 /obj/item/clothing/mask/ai/Initialize()
 	. = ..()
-	eye = new(src)
-	eye.name_sufix = "camera MIU"
-
-/obj/item/clothing/mask/ai/Destroy()
-	if(eye)
-		if(active)
-			disengage_mask(eye.owner)
-		qdel(eye)
-		eye = null
-	..()
+	set_extension(src, /datum/extension/eye/cameranet)
 
 /obj/item/clothing/mask/ai/attack_self(var/mob/user)
 	if(user.incapacitated())
 		return
-	active = !active
-	to_chat(user, "<span class='notice'>You [active ? "" : "dis"]engage \the [src].</span>")
-	if(active)
-		engage_mask(user)
-	else
-		disengage_mask(user)
-
-/obj/item/clothing/mask/ai/equipped(var/mob/user, var/slot)
-	..(user, slot)
-	engage_mask(user)
-
-/obj/item/clothing/mask/ai/dropped(var/mob/user)
-	..()
-	disengage_mask(user)
-
-/obj/item/clothing/mask/ai/proc/engage_mask(var/mob/user)
-	if(!active)
-		return
 	if(user.get_equipped_item(slot_wear_mask) != src)
```

Comment:
So now the equip-check is back but what if the mob or someone else (via stripping, magic spells, etc.) remove the mask?

Reply:
Events hooked to the unlook proc in the extension already check for movement of both the holder and eye player, which should be sufficient for any removal of the mask, unless there's an edge case I'm not considering.